### PR TITLE
Feat/content array with pictograms

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
### Related Ticket(s)
Relative to ticket #703 

### Description

Improved on the content array with pictograms element, name was changed to `PictogramArray`.
The Storybook story was improved as well.

Screen Shot 2019-11-27 at 13 51 52

### Changelog

**Changed**

- Renamed from `ContentArrayWithPictograms` to `PictogramArray`.
- Storybook story improved.

### CODE REVIEW

File: packages/patterns-react/package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "@carbon/ibmdotcom-react": "^1.1.1",
     "@carbon/ibmdotcom-styles": "1.1.1",
     "@carbon/ibmdotcom-utilities": "1.1.1",
+    "@carbon/pictograms-react": "^10.7.0",
```
Comment (jeffchew) [inline packages/patterns-react/package.json]: @RaphaelAmadeu we can put this under `devDependencies` now and not `dependencies`

File: packages/patterns-react/src/patterns/PictogramArray/__stories__/PictogramArray.stories.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { DDS_PICTOGRAMARRAY } from '../../../internal/FeatureFlags';
+import { withKnobs, text, object, select } from '@storybook/addon-knobs';
+import '../../../../../styles/scss/patterns/pictogramarray/index.scss';
+import PictogramArray from '../PictogramArray';
+import { Desktop, Touch, Pattern } from '@carbon/pictograms-react';
+// import readme from '../README.md';
```
[the remark this answers is not in the document]
Comment (annawen1) [reply packages/patterns-react/src/patterns/PictogramArray/__stories__/PictogramArray.stories.js]: yes let's include a readme, we also need another readme for the scss

File: packages/patterns-react/src/internal/FeatureFlags.js [hunk 3]
```diff
@@ -39,3 +39,11 @@ export const LISTSECTION =
  */
 export const SIMPLELONGFORM =
   process.env.SIMPLELONGFORM === 'true' || DDS_FLAGS_ALL || false;
+
+/**
+ * This determines if the contentarraywithpictograms will be rendered or not
+ *
+ * @type {string | boolean}
+ */
+export const DDS_PICTOGRAMARRAY =
```
[the remark this answers is not in the document]
Comment (annawen1) [reply packages/patterns-react/src/internal/FeatureFlags.js]: I prefer separating the words - easier to read

[the remark this answers is not in the document]
Comment (jeffchew) [reply packages/patterns-react/src/internal/FeatureFlags.js]: Agreed.

File: packages/styles/scss/ibm-dotcom-styles.scss [hunk 4]
```diff
@@ -27,3 +27,4 @@
 @import 'patterns/leadspace/leadspace';
 @import 'patterns/listsection/listsection';
 @import 'patterns/simplelongform/simplelongform';
+@import 'patters/pictogramarray/pictogramarray';
```
Comment (jeffchew) [inline packages/styles/scss/ibm-dotcom-styles.scss]: @RaphaelAmadeu misspelled `patterns` folder. Also more nitpicking, can the imports be in alphabetical order?

File: packages/patterns-react/src/patterns/index.js [hunk 5]
```diff
@@ -8,3 +8,4 @@
 export * from './LeadSpace';
 export * from './ListSection';
 export * from './SimpleLongForm';
+export * from './PictogramArray';
```
Comment (jeffchew) [inline packages/patterns-react/src/patterns/index.js]: minor nitpick, can this be in alphabetical order with the other exports?

Comment (RaphaelAmadeu) [reply packages/patterns-react/src/patterns/index.js]: I've ordered it properly in the last commit.

File: packages/styles/scss/ibm-dotcom-styles.scss (hunk 4 above)
Comment (RaphaelAmadeu) [reply packages/styles/scss/ibm-dotcom-styles.scss]: I've ordered it properly in the last commit.

File: packages/patterns-react/src/patterns/PictogramArray/__stories__/PictogramArray.stories.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (RaphaelAmadeu) [reply packages/patterns-react/src/patterns/PictogramArray/__stories__/PictogramArray.stories.js]: Both readme files were created and included on the last commit.
Thanks!

File: packages/patterns-react/src/patterns/PictogramArray/README.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+```javascript
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { PictogramArray } from '@carbon/ibmdotcom-patterns-react';
+import { Desktop, Touch, Pattern } from '@carbon/pictograms-react';
+import 'yourapplication.scss';
+import '@carbon/ibmdotcom-styles/scss/patterns/pictogramarray/index.scss';
+
+function App() {
+  return;
```
Comment (jeffchew) [inline packages/patterns-react/src/patterns/PictogramArray/README.md]: @RaphaelAmadeu minor thing, might help in the example to define `title` and `contentGroup`.

Comment (RaphaelAmadeu) [reply packages/patterns-react/src/patterns/PictogramArray/README.md]: @jeffchew I've included this in the latest version. Thanks!

Comment (jeffchew) [reply packages/patterns-react/src/patterns/PictogramArray/README.md]: I meant something like this:

```javascript
import { Pattern } from '@carbon/pictograms-react';

function App() {
  const title = 'Lorem Ipsum';
  const contentGroup = {[ 
    {
      title: 'Lorem', 
      copy: '', 
      link: {
        href: 'https://www.ibm.com', 
        text: 'IBM', 
        target: '_self'
      }, 
      pictogram: Pattern
    }
  ]};
  <PictogramArray title={title} contentGroup={contentGroup} />;
 };
}
```

Also, that `return;` before the `<PictogramArray>` doesn't seem to make sense.

**Comment by jeffchew** [comment]
@RaphaelAmadeu your yarn lock file might be out of sync, as well as your yarn offline mirror. Try running a `yarn reset` on root then push up.

**Pull request merged**